### PR TITLE
HPCC-27631 Fix error 'createIOStreamWithFileName' not declared

### PR DESCRIPTION
**Pull request by wangkx**
A call to createIOStreamWithFileName() is added by
https://github.com/hpcc-systems/HPCC-Platform/pull/15992 (merged to candidate-8.6.x repo).
But, in the master repo, the createIOStreamWithFileName()
has been replaced by createIOStreamFromFile() (https://github.com/hpcc-systems/HPCC-Platform/pull/15974).

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 1]
```diff
@@ -4108,7 +4108,10 @@ void CWsWuFileHelper::sendWUComponentLogStreaming(CHttpRequest* request, CHttpRe
                 (opt == CWUFileDownloadOption_GZIP) ? "tar" : "zip");
 
         //Send the zipped file and clean the workingFolder.
-        response->setContent(createIOStreamWithFileName(zipFileNameWithPath, IFOread));
+        auto stream = createIOStreamFromFile(zipFileNameWithPath, IFOread);
```
Comment (afishbeck) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp:4111]: I'm not a huge fan of auto when it's just used as a shortcut for a known type.  It's very helpful for template based and other obscure return values though.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 2]
```diff
@@ -4108,7 +4108,10 @@ void CWsWuFileHelper::sendWUComponentLogStreaming(CHttpRequest* request, CHttpRe
...
+        if (stream == nullptr)
+            throw makeStringExceptionV(ECLWATCH_INTERNAL_ERROR, "Cannot open stream for zip file %s", zipFileNameWithPath.str());
+        response->setContent(stream);
```
Comment (afishbeck) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp:4114]: Does this take ownership and free the stream?, or will this leak?

Comment (afishbeck) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp:4114]: Never mind, I see it does take ownership.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (wangkx) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp:4111]: Sorry, I copied from https://github.com/hpcc-systems/HPCC-Platform/pull/15974. I will change it to IFileIOStream*.

**Review by afishbeck — APPROVED** [review]

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Comment by wangkx** [comment]
@ghalliday please merge this PR.

**Comment by afishbeck** [comment]
@ghalliday note that this fixes a container mode build break.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**